### PR TITLE
feat(admin-ui): upload files to ipfs

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(admin-ui): upload files to ipfs
 
## Summary:
IPFS upload with heliaJS -> only issue its throwing error for pinging web-socket server (?!?) 
Screenshot 2024-04-04 at 18 59 37

@chefsale @xilosada 
Fixes # C-20-37

## Test plan:

https://github.com/calimero-is-near/cali2.0-experimental/assets/93442516/5378e5c2-9fc4-4094-a271-7d8447a07cd5

### CODE REVIEW

File: node-ui/src/components/uploadApp/UploadAppContent.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  padding: 42px 26px 26px 26px;
+  display: flex;
+  flex: 1;
+  .content-card {
+    background-color: #353540;
+    border-radius: 4px;
+    padding: 30px 26px 26px 30px;
+    width: 100%;
+  }
+
```
Comment (xilosada) [inline node-ui/src/components/uploadApp/UploadAppContent.jsx:17]:
```suggestion
```

File: node-ui/src/pages/UploadApp.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const { cidString, commitWasm, fetchWasm } = useUploadFile();
+  const [wasmFile, setWasmFile] = useState();
+  const [isUploaded, setIsUploaded] = useState(false);
+  const [downloadUrl, setDownloadUrl] = useState();
+
+  const handleFileChange = (event) => {
+    const file = event.target.files[0];
+    if (file && file.name.endsWith(".wasm")) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
```
Comment (xilosada) [inline node-ui/src/pages/UploadApp.jsx:19]: Error handling?

File: node-ui/src/provider/HeliaProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    if (helia) {
+      console.info("helia already started");
+    } else if (window.helia) {
+      console.info("found a windowed instance of helia, populating ...");
+      setHelia(window.helia);
+      setFs(unixfs(helia));
+      setStarting(false);
+    } else {
+      try {
+        console.info("Starting Helia");
```
Comment (xilosada) [inline node-ui/src/provider/HeliaProvider.jsx]:
```suggestion
```

File: node-ui/src/components/uploadApp/UploadAppContent.jsx (hunk 1 above)
Comment (frdomovic) [reply node-ui/src/components/uploadApp/UploadAppContent.jsx:17]: Added space at the top so its consistent across files

File: node-ui/src/provider/HeliaProvider.jsx (hunk 3 above)
Comment (frdomovic) [reply node-ui/src/provider/HeliaProvider.jsx]: removed

File: node-ui/src/pages/UploadApp.jsx (hunk 2 above)
Comment (frdomovic) [reply node-ui/src/pages/UploadApp.jsx:19]: added render.onerror

**Pull request merged**